### PR TITLE
[12.0][IMP] sale_margin_delivered: No compute margin delivered fields for no delivered lines

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT31673

ping @chienandalu

### CODE REVIEW

File: sale_margin_delivered/tests/test_sale_margin_delivered.py [hunk 1]
```diff
@@ -60,10 +60,9 @@ def _new_sale_order(self):
     def test_sale_margin_ordered(self):
         sale_order = self._new_sale_order()
         order_line = sale_order.order_line[:1]
-        self.assertEqual(order_line.margin_delivered, 60.0)
-        self.assertEqual(order_line.margin_delivered_percent, 50.0)
-        self.assertEqual(
-            order_line.purchase_price_delivery, order_line.purchase_price)
+        self.assertEqual(order_line.margin_delivered, 0.0)
```
Comment (chienandalu) [inline sale_margin_delivered/tests/test_sale_margin_delivered.py:63]: So the test results were forced to fit the result? :sweat_smile:

**Review by chienandalu — APPROVED** [review]

**Comment by sergio-teruel** [comment]
Add migration scripts.

File: sale_margin_delivered/tests/test_sale_margin_delivered.py (hunk 1 above)
Comment (sergio-teruel) [reply sale_margin_delivered/tests/test_sale_margin_delivered.py:63]: it's a change in behavior.

**Review by pedrobaeza — APPROVED** [review]
Understood the problem

**Comment by pedrobaeza** [comment]
It seems a random error:

/ocabot merge major

**Comment by pedrobaeza** [comment]
I give up. There's an error that is fluctuating between the 2 branches, and sometimes there's only one and other times 2. @sergio-teruel maybe you can rebase or inspect current ones to see what is going on.

**Comment by sergio-teruel** [comment]
Rebase done... but I do not see any change..

**Comment by pedrobaeza** [comment]
It seems you haven't pushed such rebase to the branch.

File: sale_margin_delivered/__manifest__.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 {
     "name": "Sale Margin Delivered",
-    "version": "12.0.1.0.1",
+    "version": "12.0.1.0.2",
```
Comment (dreispt) [inline sale_margin_delivered/__manifest__.py:5]: No need to do this, as ocabot can do the version bumping upon merge

Comment (chienandalu) [reply sale_margin_delivered/__manifest__.py:5]: As there's a migration script, it's forced to ensure the right path

**Pull request merged**